pulpcore/cli/config.py

```diff
@@ -0,0 +1,35 @@
+import os
+
+import click
+import toml
+
+
+DEFAULT_PATH = os.path.join(
+    os.path.expanduser(os.environ.get("XDG_CACHE_HOME") or "~/.config"), "pulp", "settings.toml"
+)
+
+
+def _config_callback(ctx, param, value):
+    if ctx.default_map:
+        return
+
+    if value:
+        ctx.default_map = toml.load(value)["cli"]
+    else:
+        try:
+            ctx.default_map = toml.load(DEFAULT_PATH)["cli"]
+        except FileNotFoundError:
+            pass
+
+
+def config_file():
+    def decorator(f):
+        options = {
+            "type": click.Path(resolve_path=True),
+            "help": "Path to the toml config file",
+            "callback": _config_callback,
+            "expose_value": False,
+        }
+        return click.option("--config", **options)(f)
```

Comment:
Why assemble the options into a dict instead of just passing them?
Also something like
`config_file = click.option("--config", ...)`
should be the decorator you are looking for.

Reply:
Originally I was going to add options to config_file(). But yes your suggestion would be simpler. My other thought though is perhaps we don’t even need a decorator. Maybe I should just add a config option to main. Wdyt?

Reply:
I've updated this PR to not use a decorator but am happy to define the `config_file` decorator if you think that's a better solution.

Reply:
Just adding config_file to main seems to be the most simple solution.